ott/core/gw_barycenter.py

```diff
@@ -0,0 +1,323 @@
+from functools import partial
+from typing import Any, Dict, NamedTuple, Optional, Sequence, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+
+from ott.core import (
+    fixed_point_loop,
+    gromov_wasserstein,
+    problems,
+    quad_problems,
+    was_solver,
+)
+from ott.core.bar_problems import GWBarycenterProblem
+from ott.geometry import geometry, pointcloud
+
+__all__ = ["GWBarycenterState", "GromovWassersteinBarycenter"]
+
+
+class GWBarycenterState(NamedTuple):
+  """Holds the state of the :class:`ott.core.bar_problems.GWBarycenterProblem`.
+
+  Args:
+    x: Barycenter features of shape ``[B, D_f]``. Only used in the fused case.
```

Comment:
here just a comment on the fact that B sounds like batch, whereas I feel k could be a bit more unified (this is what is used in continuous_barycenter)

Reply:
Good point, will change it to `k`. Generally do you prefer for the dim names to be lower case or upper case?